### ISSUE TITLE
Question about loss_mask behavior when mask_observations=False in _compose_final_output

### ISSUE THREAD

**Issue by liyu199809** [TIGER-AI-Lab/verl-tool#116]
Hi there,

I'm examining the `_compose_final_output` method in `AgentActorManager`, and I have a question about the branch where `self.config.mask_observations` is `False` (referencing lines around 1012-1013 in verl_tool/llm_agent/manager.py）
_compose_final_output).

```python
        # Create observation mask
        if self.config.mask_observations:
            final_output['loss_mask'] = torch.cat([
                torch.zeros_like(left_side['input_ids']),
                self.tensor_fn.create_attention_mask(final_output['responses_with_loss_mask'])
            ], dim=1) # [bs*n, prompt_length + max_response_length]
        else:
              # NOTE: training usr prompt？
            final_output['loss_mask'] = final_output['attention_mask']
        
        response_length = final_output['responses'].shape[1]
        final_output['response_mask'] = final_output['loss_mask'][:, -response_length:] 
```

In this branch, `final_output['loss_mask']` is assigned to `final_output['attention_mask']`. Since `attention_mask` is constructed by concatenating the attention mask for the *prompt* (`left_side['input_ids']`) and the attention mask for the *responses*, this means the `loss_mask` will include tokens from the prompt.

My confusion is: 
Does this imply that **gradients will be computed for prompt tokens during training**? Typically, prompts are treated as fixed input (not targets for training), and we only want to optimize the model’s loss on generated response tokens. Including prompt tokens in the `loss_mask` (thus computing gradients for them) seems inconsistent with that expectation.

Is this behavior intended, or might there be a subtle mistake here?

Thank you for your help!

**Comment by jdf-prog** [comment]
I have pushed this fix to the main branch. Please check it out!
